src/services/media-library-service.js

```diff
@@ -23,25 +23,29 @@ class MediaLibraryService {
     return this.mediaLibraryItemStore.getAllMediaLibraryItems();
   }
 
-  async getSearchableMediaLibraryItemsByTags({ query, resourceTypes }) {
-    const tagQuery = createTagSearchQuery(query);
-    if (!tagQuery.isValid) {
+  async getSearchableMediaLibraryItemsByTagsOrName({ query, resourceTypes }) {
+    const textQuery = createTextSearchQuery(query, ['tags', 'name']);
+    if (!textQuery.isValid) {
       return [];
     }
 
-    const queryConditions = [tagQuery.query, { resourceType: { $in: resourceTypes } }];
-    const mediaLibraryItems = await this.mediaLibraryItemStore.getMediaLibraryItemsByConditions(queryConditions);
+    const textAndResourceTypeQueryConditions = [textQuery.query, { resourceType: { $in: resourceTypes } }];
+    const mediaLibraryItems = await this.mediaLibraryItemStore.getMediaLibraryItemsByConditions(textAndResourceTypeQueryConditions);
+
+    const positiveTokensArray = [...textQuery.positiveTokens].filter(token => token.toLowerCase());
     return mediaLibraryItems
-      .map(item => ({
-        ...item,
-        relevance: item.tags.filter(tag => tagQuery.positiveTokens.has(tag.toLowerCase())).length
-      }))
-      .sort(by(item => item.relevance).thenBy(item => item.url));
+      .map(item => {
+        const exactTagMatchCount = item.tags.filter(tag => textQuery.positiveTokens.has(tag.toLowerCase())).length;
+        const partialNameMatchCount = positiveTokensArray.filter(token => item.name.toLowerCase().includes(token)).length;
+        const relevance = exactTagMatchCount + partialNameMatchCount;
+        return { ...item, relevance };
+      })
+      .sort(by(item => item.relevance).thenBy(item => item.name));
   }
```

Comment:
Maybe here we could benefit from a little test with a single set of data where:
- tags match exactly
- tags match partially
- name matches exactly
- name matches partially in the middle
- name matches partially from the beginning
so that if we need to adjust the matching, we can have specific tests failing.
What do you think?

Reply:
That's true. Although we already test the `createTextSearchQuery` separately, it's still a nice documentation in case of future changes, also we can test the correct value for relevance. Added!

Reply:
Very thorough cases, nice!